app/src/org/commcare/utils/FirebaseMessagingUtil.java

```diff
@@ -498,6 +499,10 @@ public static Intent getIntentForPNIfAny(Context context,Intent intent){
         return null;
     }
 
+    public static Intent getIntentForPNClick(Context context, PushNotificationRecord pushNotificationRecord){
+        return cccCheckPassed(context) ? handleNotification(context,PushNotificationApiHelper.INSTANCE.convertPNRecordToPayload(pushNotificationRecord),null ,false): null;
```

Comment:
nit: rename `handleNotification` to `getIntentForNotification` to make it more clear that it's returning an intent that should be handled by caller.